models/resnet.py

```diff
@@ -27,14 +27,16 @@ def __call__(self, x):
 
 
 class Bottleneck:
-  # NOTE: the original implementation places stride at the first convolution (self.conv1), this is the v1.5 variant
+  # NOTE: stride_in_1x1=False, this is the v1.5 variant
   expansion = 4
 
-  def __init__(self, in_planes, planes, stride=1, groups=1, base_width=64):
+  def __init__(self, in_planes, planes, stride=1, stride_in_1x1=False, groups=1, base_width=64):
     width = int(planes * (base_width / 64.0)) * groups
-    self.conv1 = nn.Conv2d(in_planes, width, kernel_size=1, bias=False)
+    # NOTE: the original implementation places stride at the first convolution (self.conv1), control with stride_in_1x1
+    stride1x1, stride3x3 = (stride, 1) if stride_in_1x1 else (1, stride)
```

Comment:
Err, are you sure this is right?

Reply:
yes, the reference uses the  old resnet https://github.com/mlcommons/training/blob/master/object_detection/pytorch/maskrcnn_benchmark/config/defaults.py#LL277C2-L277C38
moreover, until I added this my model outputs looked quite different compared to reference

Reply:
Oh...I'm just confused by this tuple assignment. Don't do it like this, instead do it on the lines

self.conv1 ... `stride if stride_in_1x1 else 1`
self.conv2 ... `1 if stride_in_1x1 else stride`

Reply:
changed